sqlmesh/utils/date.py

```diff
@@ -273,3 +274,15 @@ def time_like_to_str(time_like: TimeLike) -> str:
     if is_date(time_like):
         return to_ds(time_like)
     return to_datetime(time_like).isoformat()
+
+
+def to_end_date(
+    end_and_units: t.Union[t.Tuple[int, IntervalUnit], t.List[t.Tuple[int, IntervalUnit]]]
+) -> TimeLike:
+    from sqlmesh.core.model.meta import IntervalUnit
+
+    end_and_units = [end_and_units] if isinstance(end_and_units, tuple) else end_and_units
```

Comment:
I thought there was `ensure_list` function somewhere.

Reply:
This is a confusing nuance of that function. If you pass it a tuple then it will actually return a list of the tuple values instead of a list with a single value being the tuple. I want a list with the tuple being a value so I do this here.